### ISSUE TITLE
haskellPackages.idris broken after megaparsec update

### ISSUE THREAD

**Issue by tg-x** [NixOS/nixpkgs#53406]
## Issue description

`idris-1.3.1` fails to build after [`megaparsec` was updated](https://github.com/NixOS/nixpkgs/commit/11312448890631c90b793cce60765be9a65049ed),
since it requires `megaparsec >= 6.2 && < 7` but nix tries to build it with `megaparsec-7.0.4`

See also related upstream [issue](https://github.com/idris-lang/Idris-dev/issues/4551) & [PR](https://github.com/idris-lang/Idris-dev/pull/4552/files)

## Build log

```
Configuring idris-1.3.1...
...
Dependency megaparsec -any: using megaparsec-7.0.4
...
```

```
[ 59 of 115] Compiling Idris.Parser.Stack ( src/Idris/Parser/Stack.hs, dist/build/Idris/Parser/Stack.o )

src/Idris/Parser/Stack.hs:70:29: error:
    Not in scope: ‘P.errorPos’
    Module ‘Text.Megaparsec’ does not export ‘errorPos’.
   |
70 |       (pos NonEmpty.:| _) = P.errorPos err
   |                             ^^^^^^^^^^

src/Idris/Parser/Stack.hs:76:34: error:
    Not in scope: ‘P.parseErrorPretty'’
    Perhaps you meant one of these:
      ‘P.parseErrorPretty’ (imported from Text.Megaparsec),
      ‘P.parseErrorTextPretty’ (imported from Text.Megaparsec)
    Module ‘Text.Megaparsec’ does not export ‘parseErrorPretty'’.
   |
76 | prettyError (ParseError s err) = P.parseErrorPretty' s err
   |                                  ^^^^^^^^^^^^^^^^^^^

src/Idris/Parser/Stack.hs:107:30: error:
    Not in scope: ‘P.getPosition’
    Module ‘Text.Megaparsec’ does not export ‘getPosition’.
    |
107 | getFC = sourcePositionFC <$> P.getPosition
    |                              ^^^^^^^^^^^^^
builder for '/nix/store/a0ha7yjcb2zbbpz2jq8m2b0qk7q2cr4n-idris-1.3.1.drv' failed with exit code 1
```

### Steps to reproduce

`nix-build -A haskellPackages.idris`

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

``
 - system: `"x86_64-linux"`                                                                                                                                                                                                       
 - host os: `Linux 4.14.89, NixOS, 18.09.git.3843c93 (Jellyfish)`                                                                                                                                                                 
 - multi-user?: `yes`                                                                                                                                                                                                             
 - sandbox: `yes`                                                                                                                                                                                                                 
 - version: `nix-env (Nix) 2.1.3`                                                                                                                                                                                                                                                                                                                                                                
```

**Comment by tg-x** [comment]
cc @peti @Infinisil

**Comment by Infinisil** [comment]
Megaparsec 7 support has been merged: https://github.com/idris-lang/Idris-dev/pull/4610

Meanwhile I submitted a PR to fix this temporarily

**Comment by Infinisil** [comment]
Once the Haskell packages are updated, I can make a PR to fix the idris build

**Comment by Infinisil** [comment]
The fix is now in master :)

**Comment by fabianhjr** [comment]
This issue seems to have reappeared with idris-1.3.2 and it is marked as broken on nixos-unstable.

**Comment by Infinisil** [comment]
Thanks for the report, fixed in 382786179926abebd1ba6bc3bc4b6451da4993de which will get to master eventually

**Comment by fabianhjr** [comment]
Yay, thanks for the prompt patch. ^_^
